network/tor/tor/src/main/java/bisq/network/tor/TorService.java

```diff
@@ -141,15 +141,30 @@ public CompletableFuture<Boolean> initialize() {
         PasswordDigest hashedControlPassword = PasswordDigest.generateDigest();
         Map<String, String> torrcConfigMap = createTorrcConfigFile(torDataDirPath, hashedControlPassword);
 
+        // remove the control port file before we launch tor, just in case it exists from
+        // a previous run
+        Path controlDirPath = torDataDirPath.resolve(BaseTorrcGenerator.CONTROL_DIR_NAME);
+        Path controlPortFilePath = controlDirPath.resolve("control");
+        try {
+            FileUtils.deleteFile(controlPortFilePath.toFile());
```

Comment:
For a user upgrading the file exists, but the delete might be delayed a bit by the OS and the poller might then read the old file. To not depend on the pollers resiliance handling, I think its better the wait in a loop for the state that the file does not exist anymore in case the file existed before. E.g. a while loop with a 100 ms sleep and check that the previously existing file does now not exist anymore. Then we can be sure that the poller cannot read a stale file.